pkg/operator/controller/vulnerabilityreport.go

```diff
@@ -73,13 +74,13 @@ func (r *VulnerabilityReportReconciler) SetupWithManager(mgr ctrl.Manager) error
 			return err
 		}
 	}
+	var predicates []predicate.Predicate
+	if !r.ConfigData.IsScanJobsInSameNamespace() {
+		predicates = append(predicates, InNamespace(r.Config.Namespace))
+	}
+	predicates = append(predicates, ManagedByStarboardOperator, IsVulnerabilityReportScan, JobHasAnyCondition)
```

Comment:
reason for doing other way round, so that InNamespace predicate will get hit before other predicates. If the sequence of the predicates is not guaranteed then we can go ahead with your suggested way.